### ISSUE TITLE
[Accessibility]: Code Insights Dashboards - Find specific insight

### ISSUE THREAD

**Issue by unclejustin** [sourcegraph/sourcegraph#41680]
### Audit type

Keyboard navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/33501

### Problem description

The current method to find a specific insight is to scroll and look for it in a dashboard, clicking the actions menu for that insight card.

It is possible to scroll with the keyboard and find your insight, but selecting that insight is problematic.

Every series and point on a series is an interactive item. That means you must tab through every insight, every series within that insight, and every point within that series to get to your insight.

This can easily be over 100 tab presses 😬 

About keyboard navigation related comment from Felix Becker https://github.com/sourcegraph/sourcegraph/issues/33492#issuecomment-1266125114

### Expected behavior

Tabbing should cycle through insight cards. A card should then give the option to tab within that card.

### Additional details

This issue https://github.com/sourcegraph/sourcegraph/issues/41666 is about adding a filter for insights which may also fix this problem. If we can limit to only showing the insight you want, then the tabbing becomes irrelevant for this case.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

/cc @joelkw @felixfbecker @vovakulikov

**Comment by felixfbecker** [comment]
Since the dashboard is a grid, I think it should be an ARIA `grid` role with two-dimensional arrow key navigation. That would also help non-screen reader keyboard users

**Comment by vovakulikov** [comment]
@felixfbecker 
> Since the dashboard is a grid

I think this isn't quite correct (or I just think in a wrong way about this). Our dashboard doesn't have any rows, in fact the dashboard page content is more like a wrapped list of cards instead of grid. The spec says that grid should contains rows element and rows must contain grid cells. 

>Authors MUST ensure elements with role gridcell, columnheader, or rowheader are [owned](https://www.w3.org/TR/wai-aria-1.1/#dfn-owned-element) by elements with role [row](https://www.w3.org/TR/wai-aria-1.1/#row), which are in turn owned by an element with role [rowgroup](https://www.w3.org/TR/wai-aria-1.1/#rowgroup), or grid.

So since we don't have any rows (we may have them visually but semantically we don't) I'm not sure that grid here is appropriate way to set roles. I agree about keyboard navigation this part I believe we should implement for the dashboard cards. 

There is a good overview on this problem [here](https://sarahmhigley.com/writing/grids-part1/). There are lines about grid definition 

> Highly interactive sets of items that do not have meaningful row and column relationships are also not grids. This is still true even if they are visually laid out in rows and columns, if those rows and columns do not provide meaning. A good example is this interface that allows users to pick from a list of apps: it is visually laid out in a grid pattern, but is not semantically a grid

So at the end I think we should have something in between grids and interactive list of elements. Our columns and rows are just visual attributes it makes me think that we should go with list role here and implement keyboard naviagation as it's described in grid specification but don't use grid-like roles.

Feel free to disagree, curios about your opinion on this @felixfbecker @unclejustin.

**Comment by umpox** [comment]
Interesting problem!

> Highly interactive sets of items that do not have meaningful row and column relationships are also not grids. This is still true even if they are visually laid out in rows and columns, if those rows and columns do not provide meaning. A good example is this interface that allows users to pick from a list of apps: it is visually laid out in a grid pattern, but is not semantically a grid

I was thinking that drag-and-drop behavior might give these row/column relationships more meaning, but I actually I still think that's not the case. This page just collapses into a single column view on the smallest screens, so any columns that might have been implied are redundant anyway.

I did some more digging and the clearest example I could find that matches this UI is [scrollable search results (W3)](https://www.w3.org/WAI/ARIA/apg/example-index/grid/LayoutGrids.html#ex3_label). However this article: [ARIA Grid As an Anti-Pattern](https://adrianroselli.com/2020/07/aria-grid-as-an-anti-pattern.html#Layout) goes into detail about why layout grids aren't necessarily a good choice, (and additional considerations that you need to account for). I also feel like making this a `list` (with typical tab navigation) is the best way forward. We can still enhance this with arrow key navigation afterwards
